collectors/30/kafka_offset_collector.py

```diff
@@ -16,103 +21,166 @@ def format_tsd_key(metric_key, metric_value, timestamp, tags={}):
+    while True:
+        try:
+            children = zk.get_children(partition_path)
+        except NoNodeError:
+            return 0
+        try:
+            batch_number = max(map(int, children))
+            batch_path = StormPaths.batch(partition, batch_number)
+            batch_info = json.loads(zk.get(batch_path)[0])
+            return batch_info['offset']
+        except NoNodeError:
+            # the batches are changing frequently so this will happen
+            continue
+
+
 # KAFKA OFFSET MONITORING
 
 
-def get_partitions(zk, group, topic):
-    return zk.get_children(KafkaPaths.topic_offsets(group, topic))
+def get_partitions(zk, topic):
+    path = KafkaPaths.partitions(topic)
+    return map(int, zk.get_children(path))
+
+
+def get_kafka_topics(zk):
+    path = KafkaPaths.topics()
+    return [str(topic) for topic in zk.get_children(path) if topic not in BLACKLIST]
 
 
 def get_consumer_group_offset(zk, group, topic, partition):
+    if group == 'storm':
+        # special case it
+        return get_storm_offset(zk, partition)
     path = KafkaPaths.consumer_topic_partition(group, topic, partition)
-    return int(zk.get(path)[0])
-
+    try:
+        return int(zk.get(path)[0])
+    except NoNodeError:
+        return 0
 
-def report_broker_info(kafka, zk, topic):
-    def get_partitions(topic):
-        path = KafkaPaths.broker_partitions(topic)
-        return map(int, zk.get_children(path))
 
+def report_broker_info(env, kafka, zk, topic):
     total_offset = 0
-    for partition in get_partitions(topic):
+    for partition in get_partitions(zk, topic):
         offset_response = kafka.send_offset_request([OffsetRequest(topic, partition, -1, 1)])[0]
         offset = offset_response.offsets[0]
         total_offset += offset
-        print format_kafka_broker_offset_tsd_key(topic, partition, offset)
+        print format_kafka_broker_offset_tsd_key(env, topic, partition, offset)
     return total_offset
 
 
-def report():
-    # see the monitoring.json chef role
-    zk_quorums = os.getenv('MONITORED_ZOOKEEPER_QUORUMS')
-    kafka_brokers = os.getenv('KAFKA_BROKERS')
+def add_dns_domain(hosts):
+    def add_dns_domain_to_host(host):
+        port = None
+        if ':' in host:
+            host, port = host.split(':', 1)
+        if '.' not in host:
+            host += DNS_DOMAIN
+        return host + ':' + port if port else host
+    return ','.join(map(add_dns_domain_to_host, hosts.split(',')))
 
-    if zk_quorums is None:
-        raise RuntimeError('MONITORED_ZOOKEEPER_QUORUMS not found')
 
-    if kafka_brokers is None:
-        raise RuntimeError('KAFKA_BROKERS not found')
+def parse_kafka_info(kafka):
+    # 1=2,3=4 --> {1:2, 3:4}
+    info = dict(part.split('=') for part in kafka.split(' '))
+    if 'env' not in info or 'kafka' not in info or 'zk' not in info:
+        raise RuntimeException('Kafka info must contain "env", "kafka", and "zk"')
+    zk = KazooClient(hosts=add_dns_domain(info['zk']))
```

Comment:
This is the location in which we start pointing at some different zk but it's hard to see which one...